### DISCUSSION TITLE
a column with all values being 'duckdb'

### DISCUSSION THREAD

**Discussion by eijoac** [duckdb/duckdb#6980] [Q&A] [answered]
I imported a parquet file into a table. One column has all values as 'duckdb' (see below). This column in fact stores dictionary-type data (or perhaps json with the invalid single quote syntax?). This can be seen if I turn the DuckDB table into a Python dataframe. (see below).

My questions are 1) what is this 'duckdb' value; 2) is there a way to load this parquet file correctly, i.e., the 'user' column can be a struct so that I can query the value of each key.

Thanks.

```
>>> duckdb.sql("SELECT user FROM ibm LIMIT 10;")
┌─────────┐
│ user()  │
│ varchar │
├─────────┤
│ duckdb  │
│ duckdb  │
│ duckdb  │
│ duckdb  │
│ duckdb  │
│ duckdb  │
│ duckdb  │
│ duckdb  │
│ duckdb  │
│ duckdb  │
├─────────┤
│ 10 rows │
└─────────┘
```

```
>>> test = duckdb.sql("SELECT * FROM ibm LIMIT 10;").df()
>>> test["user"].head()
0    {'id': 123, 'username': 'xyz', 'name': '...
1    {'id': 456, 'username': 'abc', 'name': 'Lei...
2    {'id': 1111, 'username': 'xyzdd', '...
3    {'id': 2222, 'username': 'abcdd', 'nam...
4    {'id': 3333, 'username': 'efg', 'name':...
Name: user, dtype: object
```

**Comment by Mytherin** [comment, marked as answer]
This is a nice easter-egg from the Postgres parser - `user` is a special "column function" - which is a column that automatically gets translated into a function call when used unquoted, for example:

```sql
D select user;
┌─────────┐
│ user()  │
│ varchar │
├─────────┤
│ duckdb  │
└─────────┘
```

To get the correct value here you need to add quotes (e.g. `SELECT "user" FROM ibm LIMIT 10`). I'm going to turn this into an issue and fix this since this is very confusing to users.

**Reply by Mytherin** [reply]
Actually I had already opened an issue for it (#6232).

**Reply by eijoac** [reply]
Thanks for the quick reply. It worked. Is there a way to easily refer to a value of a certain key in that column. The below doesn't work as I guess it's due to invalid json syntax (the single quote). Is there a workaround? Thanks.

```
>>> duckdb.sql('select "user"->>id from ibm limit 10;')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
duckdb.InvalidInputException: Invalid Input Error: Malformed JSON at byte 1 of input: unexpected character.  Input: {'id': 123, 'username': 'xyz', 'name': 'x...
```

**Reply by Mytherin** [reply]
`->>` only works with JSON columns, if `user` is not a JSON column but rather a struct you can use the `.` syntax, e.g. `"user".id`.

**Reply by eijoac** [reply]
`"user".id` won't work either as the column seems to be imported as a varchar. Is there a workaround? Thanks.

>>> duckdb.sql('select "user".id from ibm limit 10;')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
duckdb.BinderException: Binder Error: Cannot extract field 'id' from expression "user" because it is not a struct or a union

**Reply by Mytherin** [reply]
You might need to convert it to a struct explicitly, e.g:

```sql
select "user"::ROW(id INT, username VARCHAR, name VARCHAR, ...) from ibm limit 10;
```

**Reply by eijoac** [reply]
Thanks. It's a long dictionary with many keys, but will give it a try.
